src/main/java/edu/harvard/iq/dataverse/api/MxrdrWorkflowApi.java

```diff
@@ -0,0 +1,125 @@
+                        WorkflowExecution execution = result.get();
+                        if (execution.isFinished() && !execution.getLastStep().getFinishedSuccessfully()) {
+                            rerun(dataset, released);
+                            datasetProcessed++;
+                        }
+                    } else if (!result.isPresent() && type.equals("notPerformedOnly")) {
+                        rerun(dataset, released);
+                        datasetProcessed++;
+                    }
+                }
+            }
+            
+            return ok("Processed " + datasetProcessed + " datasets");
+        } catch (WrappedResponse ex) {
+            return ex.getResponse();
+        } catch (NoDatasetFilesException ex) {
+            return error(Response.Status.INTERNAL_SERVER_ERROR,
+                         "Unable to publish dataset, since there are no files in it.");
+        }
+
+    }
+
+    @POST
+    @Path("{id}/workflow/rerun")
+    public Response rerunWorkflow(@PathParam("id") String id, @QueryParam("version") String version) {
+        try {
+            AuthenticatedUser user = findAuthenticatedUserOrDie();
+            if (!user.isSuperuser()) {
+                return error(Response.Status.FORBIDDEN, "Only superusers can run workflow");
+            }
+            Dataset ds = findDatasetOrDie(id);
+            DatasetVersion updateVersion = null;
+            if (version != null) {
+                for (DatasetVersion datasetVersion:ds.getVersions()) {
+                    if (version.equals(datasetVersion.getVersionNumber() + "." + datasetVersion.getMinorVersionNumber())) {
+                        updateVersion = datasetVersion;
+                    }
+                }
+                if (updateVersion == null) {
+                    return error(Response.Status.BAD_REQUEST, "Unknown version: " + version);
+                } else {
+                    rerun(ds, updateVersion);
+                }
+            } else {
+                return error(Response.Status.BAD_REQUEST, "Missing version number");
+            }
+            
+            return ok("Datasets processed");
+        } catch (WrappedResponse ex) {
+            return ex.getResponse();
+        } catch (NoDatasetFilesException ex) {
+            return error(Response.Status.INTERNAL_SERVER_ERROR,
+                         "Unable to publish dataset, since there are no files in it.");
+        }
+
+    }
+    
+    private void rerun(Dataset dataset, DatasetVersion datasetVersion) throws WrappedResponse {
+        Optional<Workflow> workflow = workflowServiceBean.getDefaultWorkflow(PostPublishDataset);
+        if (workflow.isPresent()) {
+            workflowExecutionFacade.start(
+                    workflow.get(), new WorkflowContext(PostPublishDataset, dataset.getId(), datasetVersion.getMinorVersionNumber(), datasetVersion.getVersionNumber(), createDataverseRequest(findUserOrDie()), true));
```

Comment:
Are we sure that rerun will not cause having merged dataset fields from two executions?